src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1

```diff
@@ -36,7 +36,10 @@ Gets the default location for a provider
 #>
 function Get-ProviderLocation($provider)
 {
-	if ([Microsoft.Azure.Test.HttpRecorder.HttpMockServer]::Mode -ne [Microsoft.Azure.Test.HttpRecorder.HttpRecorderMode]::Playback)
+	$assemblies = [AppDomain]::Currentdomain.GetAssemblies() | Select-Object FullName | ForEach-Object { $_.FullName.Substring(0, $_.FullName.IndexOf(',')) }
+    if ($assemblies -notcontains 'Microsoft.Azure.Test.HttpRecorder.HttpMockServer' `
```

Comment:
I am not following this.  These are types - do you mean to get the type list from each of the loaded assemblies?

Reply:
In many places, people just use a Try-Catch to detect if it is running live or using mocking. Instead, I check to see if the assemblies are available and then check the playback mode. I had to do this or else this call throws an exception. Basically, the same check that was there before except with type safety.